src/components/screens/admin-vacation-management/EditVacationDialog.tsx

```diff
@@ -67,42 +85,84 @@ const EditVacationDialog = ({
   onSave,
   disableSave
 }: EditVacationDialogProps) => {
+  const currentYear = new Date().getFullYear().toString();
+  const[selectedYear,setSelectedYear] = useState<string>(currentYear);
+  const availableYears = generateYearOptions();
+
   if (!user) return null;
 
   const normalizedVacationDays = normalizeVacationDays(vacationDays);
-  const years = Object.keys(normalizedVacationDays).filter(
-    (year) => Number.parseInt(year) <= new Date().getFullYear()
-  );
+
+  // Get the data for the selected year, or initialize with zero
+  const selectedYearData = normalizedVacationDays[selectedYear] || { total: 0, remaining: 0 };
+
+  const handleYearChange = (year: string) => {
+    setSelectedYear(year);
+  };
 
   return (
-    <Dialog open={open} onClose={onClose} maxWidth="md" fullWidth>
+    <Dialog open={open} onClose={onClose} maxWidth="sm" fullWidth>
       <DialogTitle>
         {strings.adminVacationManagement.editTitle}: {user.firstName} {user.lastName}
       </DialogTitle>
       <DialogContent dividers>
-        {years.map((year) => (
-          <Box key={year} sx={{ mb: 3 }}>
-            <Typography variant="h6">{year}</Typography>
-            <Box sx={{ display: "flex", gap: 2, mt: 1 }}>
+        <Box sx={{ mb: 3 }}>
+          <FormControl fullWidth sx={{mb: 3}}>
+            <InputLabel id="year-select-label">{strings.adminVacationManagement.selectYear}</InputLabel>
+            <Select
+              labelId="year-select-label"
+              id="year-select"
+              value={selectedYear}
+              label="Select Year"
```

Comment:
All user facing string should be localized- see the other uses of strings.something throughout the code- add the necessary string to en.json, fi.json and strings.ts